Code/Tools/ProjectManager/Source/CreateAGemScreen.h

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#if !defined(Q_MOC_RUN)
+#include <ScreenWidget.h>
+#include <FormFolderBrowseEditWidget.h>
+#include <FormLineEditWidget.h>
+#include <FormComboBoxWidget.h>
+#include <GemCatalog/GemInfo.h>
+#include <PythonBindings.h>
+
+#include <QHash>
+#include <QApplication>
+#include <QStyleOptionTab>
+#include <QStylePainter>
+#include <QTabBar>
+#include <QTabWidget>
+#include <QButtonGroup>
+#include <QDialogButtonBox>
+#include <QRadioButton>
+#include <QPushButton>
+#include <QScrollArea>
+#include <QComboBox>
+#include <QVBoxLayout>
+#endif
+
+//#define CREATE_A_GAME_ACTIVE
+
+namespace O3DE::ProjectManager
+{
+    class TabBar : public QTabBar
+    {
+    public:
+        QSize tabSizeHint(int index) const
+        {
+            QSize s = QTabBar::tabSizeHint(index);
+            s.transpose();
+            return s;
+        }
+
+    protected:
+        void paintEvent(QPaintEvent*)
+        {
+            QStylePainter painter(this);
+            QStyleOptionTab opt;
+
+            QPoint currentTopPosition(115, 175);
+
+            QStringList strs = { "1. Gem Setup", "2. Gem Details", "3. Creator Details" };
+            for (int i = 0; i < count(); i++)
+            {
+                initStyleOption(&opt, i);
+                painter.drawControl(QStyle::CE_TabBarTabShape, opt);
+                painter.save();
+                QSize s = opt.rect.size();
+                s.transpose();
+                s.setWidth(130);
+                QRect r(QPoint(), s);
+                r.moveCenter(opt.rect.center());
+                opt.rect = r;
+                QPoint c = tabRect(i).center();
+                QPoint leftJustify(currentTopPosition);
+                leftJustify.setX(30 + (int)(0.5 * opt.rect.width()));
+                painter.translate(leftJustify);
+                currentTopPosition.setY(currentTopPosition.y() + 55);
+                painter.setFont(QFont("Open Sans", 12));
+                painter.translate(-c);
+                painter.drawItemText(r, Qt::AlignLeft, QApplication::palette(), true, strs.at(i));
+                painter.restore();
+            }
+        }
+    };
+
+    class TabWidget : public QTabWidget
+    {
+    public:
+        TabWidget(QWidget* parent = 0)
+            : QTabWidget(parent)
+        {
+            setTabBar(new TabBar);
+            setTabPosition(QTabWidget::West);
+        }
+    };
```

Comment:
I recommend either forward declare `TabWidget` and move all the declaration/implementation code out of the header and into the creategem.cpp - or make a re-usable version in a separate .h/.cpp

Reply:
I'm going to investigate not needing this at all, there should be a way of having this functionality without needing to manually paint so we can use the stylesheet

Reply:
Moved this to the cpp so the includes can be removed from the header. From a quick investigation it looks like it should be possible to remove this completely but I'll do this after the feature is in so to not hold it up much further.